### PR TITLE
Fixes #498.  Add route table option to subnet create and update.

### LINKED ISSUE #498: network: 'subnet create' should take route table
Both xplat and SDK expose it.
Fall into the category of `IT scenario`. Should do it by the 2nd partner release alone with VPN, DNS, and ATM
Make sure do it for `set`

### LINKED ISSUE #844: [Help] Argument groups no longer display correctly
Likely a side effect of PR #832.

```
Command
    az storage blob delete: Marks the specified blob or snapshot for deletion.
        The blob is later deleted during garbage collection.

        Note that in order to delete a blob, you must delete all of its
        snapshots. You can delete both at the same time with the Delete
        Blob operation.

Arguments
    --container-name -c [Required]: The container name.
    --name -n           [Required]: The blob name.

Storage Account Arguments
    --account-key                 : Storage account key. Must be used in conjunction with storage
                                    account name. Environment variable: AZURE_STORAGE_KEY.
    --account-name                : Storage account name. Must be used in conjunction with either
                                    storage account key or a SAS token. Environment variable:
                                    AZURE_STORAGE_ACCOUNT.
    --connection-string           : Storage account connection string. Environment variable:
                                    AZURE_STORAGE_CONNECTION_STRING.
    --delete-snapshots            : Required if the blob has associated snapshots.  Allowed values:
                                    include, only.
    --if-match                    : An ETag value, or the wildcard character (*). Specify this
                                    header to perform the operation only if the resource's ETag
                                    matches the value specified.
    --if-modified-since           : Alter only if modified since supplied UTC datetime
                                    (Y-m-d'T'H:M'Z').
    --if-none-match               : An ETag value, or the wildcard character (*). Specify this
                                    header to perform the operation only if the resource's ETag does
                                    not match the value specified. Specify the wildcard character
                                    (*) to perform the operation only if the resource does not
                                    exist, and fail the operation if it does exist.
    --if-unmodified-since         : Alter only if unmodified since supplied UTC datetime
                                    (Y-m-d'T'H:M'Z').
    --lease-id                    : Required if the blob has an active lease.

Storage Account Arguments
    --sas-token                   : A Shared Access Signature (SAS). Must be used in conjunction
                                    with storage account name. Environment variable:
                                    AZURE_STORAGE_SAS_TOKEN.
    --snapshot                    : The snapshot parameter is an opaque DateTime value that, when
                                    present, specifies the blob snapshot to delete.
    --timeout                     : Request timeout in seconds. Applies to each call to the service.
```

The arguments now appear in alphabetical order with potentially multiple group labels.

### PR DESCRIPTION
**Pull request by BurtBiel**
Fixes #498.  Fixes #844.
Also have a fix for help groups that keeps Arguments separate from un-indexed groups

### CODE REVIEW

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -665,24 +666,40 @@ def update_vnet(instance, address_prefixes=None):
...
 
     if network_security_group:
         subnet.network_security_group = NetworkSecurityGroup(network_security_group)
+    _set_route_table(ncf, resource_group_name, route_table, subnet)
 
     return ncf.subnets.create_or_update(resource_group_name, virtual_network_name,
                                         subnet_name, subnet)
 
-def update_subnet(instance, address_prefix=None, network_security_group=None):
+def update_subnet(instance, args, address_prefix=None, network_security_group=None, #pylint: disable=unused-argument
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py]: >args [](start = 28, length = 4)

This should just work with resource_group_name as a signature argument. Using dictionary-like access through generic args is one of the anti-patterns we sought to get rid of by switching to the "simple method" way of writing custom commands.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 2]
```diff
@@ -302,6 +302,7 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 register_cli_argument('network vnet subnet', 'address_prefix', metavar='PREFIX', help='the address prefix in CIDR format.')
 register_cli_argument('network vnet subnet', 'virtual_network_name', virtual_network_name_type)
 register_cli_argument('network vnet subnet', 'network_security_group', validator=validate_nsg_name_or_id)
+register_cli_argument('network vnet subnet', 'route_table', help='Name or ID of a route table to assiciate with the subnet.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: >assiciate [](start = 97, length = 9)

typo: associate

**Comment by BurtBiel** [comment]
@tjprescott Comments addressed.

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -665,24 +666,40 @@ def update_vnet(instance, address_prefixes=None):
...
 def create_subnet(resource_group_name, virtual_network_name, subnet_name,
-                  address_prefix='10.0.0.0/24', network_security_group=None):
+                  address_prefix='10.0.0.0/24', network_security_group=None,
+                  route_table=None):
     '''Create a virtual network (VNet) subnet
     :param str address_prefix: address prefix in CIDR format.
-    :param str network_security_group: attach with existing network security group,
-        both name or id are accepted.
+    :param str network_security_group: Name or ID of network security
+        group to assiciate with the subnet.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py]: >assiciate [](start = 17, length = 9)

typo: associate

**Comment by tjprescott** [comment]
One last typo in the help, but otherwise looks much better to me :)

---
In reply to: [245771575](https://github.com/Azure/azure-cli/pull/845#issuecomment-245771575) [](ancestors = 245771575)

**Comment by tjprescott** [comment]
:shipit:

**Pull request merged**